packages/governance/src/votes/votes.cairo

```diff
@@ -246,68 +300,55 @@ pub mod ERC20VotesComponent {
         /// To register a mint, `from` should be zero. To register a burn, `to`
         /// should be zero. Total supply of voting units will be adjusted with mints and burns.
         ///
+        /// WARNING: If voting units are a function of an underlying transferrable asset (like a
+        /// token), this function should be called every time the underlying asset is transferred to
+        /// keep the internal accounting of voting power in sync. For ERC20 and ERC721 tokens, this
+        /// is usually done using hooks.
+        ///
         /// May emit one or two `DelegateVotesChanged` events.
         fn transfer_voting_units(
             ref self: ComponentState<TContractState>,
             from: ContractAddress,
             to: ContractAddress,
             amount: u256
         ) {
-            let zero_address = Zero::zero();
             let block_timestamp = starknet::get_block_timestamp();
-            if (from == zero_address) {
-                let mut trace = self.ERC20Votes_total_checkpoints.read();
-                trace.push(block_timestamp, trace.latest() + amount);
+            if from.is_zero() {
+                let mut trace = self.Votes_total_checkpoints.deref();
+                trace.push(block_timestamp, trace.into().latest() + amount);
             }
-            if (to == zero_address) {
-                let mut trace = self.ERC20Votes_total_checkpoints.read();
-                trace.push(block_timestamp, trace.latest() - amount);
+            if to.is_zero() {
+                let mut trace = self.Votes_total_checkpoints.deref();
+                trace.push(block_timestamp, trace.into().latest() - amount);
             }
             self.move_delegate_votes(self.delegates(from), self.delegates(to), amount);
         }
 
         /// Returns the number of checkpoints for `account`.
-        fn num_checkpoints(self: @ComponentState<TContractState>, account: ContractAddress) -> u32 {
-            self.ERC20Votes_delegate_checkpoints.read(account).length()
+        fn num_checkpoints(self: @ComponentState<TContractState>, account: ContractAddress) -> u64 {
+            self.Votes_delegate_checkpoints.entry(account).length()
```

Comment:
Should we mention the changed return types for this and `checkpoints` in the changelog?

Reply:
We say we are adding `VotesComponent` and removing the old implementation. Since we are listing it as a breaking change I don't think we need to be explicit about this.